### PR TITLE
Add auto update feature

### PR DESCRIPTION
**Pull request by jess-sys**

### CODE REVIEW

**Review by super-bunny — APPROVED** [review]
Seems good

**Review by Drosaca — APPROVED** [review]
My code is so clean I love it

File: main/updater.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    return new MacUpdater(options)
+  }
+  return new AppImageUpdater(options)
+}
+
+export default async function appUpdate(): Promise<UpdateCheckResult | null> {
+  const options: AllPublishOptions = {
+    requestHeaders: {
+      Authorization: 'Basic AUTH_CREDS_VALUE',
+    },
```
Comment (jess-sys) [inline main/updater.ts:21]:
```suggestion
```

**Review by super-bunny — APPROVED** [review]
Seems better

**Review by Drosaca — APPROVED** [review]

**Pull request merged**